### ISSUE TITLE
display of used profile wrong

### ISSUE THREAD

**Issue by 0xFaul** [ansible/ansible-lint#3998]
##### Summary

when running ansible-lint the used profile in the out put is wrong:
```bash
ansible-lint --profile production
[...]
Passed with min profile: [...]
```
whereas
```bash
ansible-lint --profile min
[...]
Passed with production profile: [...]
```
##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console (paste below)
ansible-lint --version
ansible-lint 6.13.1 using ansible 2.14.3
```

- ansible installation method:  OS package (debian bookworm)
- ansible-lint installation method: OS package (debian bookworm)

##### STEPS TO REPRODUCE

run ansible-lint against any playbook with cli param `---profile production` -> check output
run ansible-lint against any playbook with cli param `---profile min` -> check output

##### Desired Behavior

show actual used profile
Possible security bugs should be reported via email to `security@ansible.com`

##### Actual Behavior
displayed profile is flipped min <-> production ...

**Comment by ssbarnea** [comment]
@cidrblock Maybe we can check on this because it seems to keep confusing users. We report the top profile that passed and people assume that is the "required" one.

If your code passed production, it will always report "Passed with production profile" even if you asked for min, as min is included in production.

**Comment by 0xFaul** [comment]
Not sure whether you got me right. I'm not talking about the `Failed: 66 failure(s), 0 warning(s) on 95 files. Profile 'production' was required, but 'min' profile passed.`

i'm talking about the ```
$ ansible-lint -c .ansible-lint --project-dir . --profile production
[...]
Read documentation for instructions on how to ignore specific rule violations.

                   Rule Violation Summary
 count tag               profile rule associated tags
     1 yaml[line-length] basic   formatting, yaml (warning)

Passed with min profile: 0 failure(s), 1 warning(s) on 94 files.
```
i explicitly pass the production profile and the output returns that it passed with min profile (and vice versa).
